### ISSUE TITLE
JOOQ/Could not find eligible property

### ISSUE THREAD

**Issue by javaboyjunior** [arnaudroger/SimpleFlatMapper#702]
Hi there

Trying to figure out what I am doing wrong without providing tons of code. I have a JOOQ generated POJO that has the following field definition:

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "farmer_id", nullable = false, precision = 20)
    public ULong getFarmerId() {
        return this.farmerId;
    }

for the following column definition:
	farmer_id SERIAL PRIMARY KEY,

I am running the following query:

        SelectQueryMapper<Farmer> farmerMapper = SelectQueryMapperFactory
                .newInstance()
                .newMapper(Farmer.class);

        List<Farmer> farmers = farmerMapper.asList(dslContext
                .select(FARMER.FARMER_ID, FARMER.LEGACY_ID, FARMER.COMPANY_NAME, FARMER.CONTACT_FIRST,
                        FARMER.CONTACT_LAST, FARMER.ADDRESS, FARMER.CITY, FARMER.STATE_ID,
                        FARMER.ZIP, FARMER.PHONE, FARMER.EMAIL, FARMER.WEBSITE,
                        FARMER.FARMER_TYPE_ID, FARMER.ACTIVE, FARMER.DB_DATE_CREATED, FARMER.DB_DATE_UPDATED,
                        FARMER_TYPE.FARMER_TYPE_ID, FARMER_TYPE.NAME, FARMER_TYPE.DB_DATE_CREATED,
                        FARMER_TYPE.DB_DATE_UPDATED)
                .from(FARMER).leftJoin(FARMER_TYPE).on(FARMER_TYPE.FARMER_TYPE_ID.eq(FARMER.FARMER_TYPE_ID))
                .orderBy(FARMER.COMPANY_NAME));
        return farmers;

and I get the following exception:

org.jooq.exception.MappingException: Could not find eligible property for 'farmer_id' on  class org.aim.core.persistence.tables.pojos.Farmer not found

I have tried adding an alias for the farmer_id to farmerId and that doesn't make any difference. I must be missing something but can't figure out what. As far as I can tell from the docs, my property and column names meet the criteria for  name matching without aliases. Any thoughts as to what I am doing wrong?
Thank you.

**Comment by arnaudroger** [comment]
Thanks for the report will have a look at it

**Comment by arnaudroger** [comment]
Is there a setter or a constructor that’s can set the farmerId?

**Comment by javaboyjunior** [comment]
First of all, thanks for the quick response!

farmerId does have a setter. Here's the full declaration:

@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
@Column(name = "farmer_id", nullable = false, precision = 20)
public ULong getFarmerId() {
    return this.farmerId;
}

public void setFarmerId(ULong farmerId) {
    this.farmerId = farmerId;
}


Thank you!

**Comment by javaboyjunior** [comment]
Update: I tried removing the join so was just selecting a flat list of farmers without inflating any foreign key POJOs and got the same error.

**Comment by arnaudroger** [comment]
That might because of the ulong could you try with just a regular long to see if that the case. I’ll might need to add the support for that. I don’t have much time this week to look at that though so on a more thorough look next week

**Comment by javaboyjunior** [comment]
Yes, it was the ULong- thanks. I can convert my code to longs easily
enough. If you want to point me in a general direction I could attempt a PR
if you like :-)

**Comment by arnaudroger** [comment]
I’ll keep the ticket open and try to add support for Uxxx

**Comment by javaboyjunior** [comment]
Thank you sir.
